### PR TITLE
feat: Add `interpolateColors` function

### PR DESCRIPTION
**Pull request by jtomaszewski**
## Description

Lets users animate colors with `react-native-reanimated` without the need of additional helper functions like [this one](https://github.com/software-mansion/react-native-reanimated/issues/181#issuecomment-528245794) or [this one](https://github.com/wcandillon/react-native-redash/blob/master/packages/core/src/Colors.ts#L136). 

Also will help avoid people from running into a problem "_why does this color doesn't animate?_ :(" like I did today in the morning in one of my projects. :)

Fixes https://github.com/software-mansion/react-native-reanimated/issues/181 .

### CODE REVIEW

**Comment by jakub-gonet** [comment]
Thanks for your contribution!
This also fixes #262.

I wonder if we could include it directly in `interpolate` wrapping it (if input are strings) or leave as a helper

**Comment by jtomaszewski** [comment]
> I wonder if we should include it directly in interpolate wrapping it or leave as a helper

Maybe it could be included directly in `interpolate`, but then the `interpolate` would need to be able to distinguish between a color output and non-color output? and so we'd need to edit the `Animated.color` (or create a new method for that) to create some kind of a "color class"?

I was thinking about it, but then decided: doing is definitely more work; maybe needs to be discussed first; and still, I think it can be done later on as improvement. (The helper would then still work.) 

For now I think merging this PR will be very helpful, as it'll make animating colors way easier, and also provide the docs describing how to do it. ( Currently there are none ;P )

Let me know if you want me to do some changes to the PR if anything blocks merging it, I'm happy to help.

**Comment by jtomaszewski** [comment]
Also, it does fix https://github.com/software-mansion/react-native-reanimated/issues/262 I think, at least the demo - you can see it being fixed after using the helper here: https://snack.expo.io/@jack-ailo/reanimatedcolorbugexample .

**Comment by jakub-gonet** [comment]
> the interpolate would need to be able to distinguish between a color output and non-color output, right? and so we'd need to edit the Animated.color (or create a new method for that) to create some kind of a "color class"?

You're right, yes. I was convinced that color is a node but it's just a few bitshifts glued together.

> Also, it does fix #262 (...)

That's also true, I mistook it for another issue with interpolating colors as strings e.g. from "#fff" to "#000" (which can also be added in some followup)

I think a minor update to docs would be awesome:
- crossreferencing `interpolateColors` in the interpolate and color node, so people can see that it exists
- adding info about [0, 255] range for colors (most users will know about that, but it won't hurt).

After those changes, this will be ready for merging.

**Comment by jtomaszewski** [comment]
Updates to docs added ;)

**Comment by wcandillon** [comment]
Not sure if this helps but we have implemented such functions in redash: https://github.com/wcandillon/react-native-redash/blob/master/packages/core/src/Colors.ts
maybe it can help to make the API a little bit more user friendly? For instance to pass colors as strings.

**Comment by jtomaszewski** [comment]
I didn't know that react-native has such function as `processColor` built in. I was afraid that parsing the colors as strings would require including a dependency, but I can see `react-native` is enough. I'll improve it in a minute. Thanks @wcandillon !

**Comment by jtomaszewski** [comment]
Actually the `interpolateColors` wasn't exported at all here. I added an example and fixed all the bugs. Now this PR is 100% merge ready IMHO.

File: react-native-reanimated.d.ts [hunk 1]
```diff
@@ -328,6 +328,16 @@ declare module 'react-native-reanimated' {
       value: Adaptable<number>,
       config: InterpolationConfig,
     ): AnimatedNode<number>;
+    export function interpolateColors(
+      animationValue: Adaptable<number>,
+      {
+        inputRange,
+        outputColorRange
+      }: {
+        inputRange: ReadonlyArray<Adaptable<number>>;
```
Comment (jakub-gonet) [inline react-native-reanimated.d.ts:337]: After adding string handling this shouldn't be `ReadonlyArray<Adaptable<number | string>>;`?

**Comment by wcandillon** [comment]
Just had a look at the diff and it looks good :)

**Comment by ekatzenstein** [comment]
interpolateColors is not exported with the v1.9.0

**Comment by jakub-gonet** [comment]
It's not, it'll be included in the next version.

**Pull request merged**